src/qurveros/spacecurve.py

```diff
@@ -249,20 +249,41 @@ def _frenet_dict_fun(x, params):
 
         self._frenet_dict_fun = _frenet_dict_fun
 
-    def set_params(self, params):
-
-        """
-        Sets the default parameters of the curve.
-        The associated elements are set to None so that the geometric
-        quantities, the robustness properties and the control fields are
-        evaluated for the new set of auxiliary parameters.
-        """
-
-        self.params = params
+def _cast_params_to_float(self, params):
+    """
+    Recursively cast all array-like leaves in params to jnp.float32.
+    Raises TypeError if casting is not possible.
+    """
+    import numpy
```

Comment:
This import statement should be removed- it is a duplicate of the one at the top of the file.